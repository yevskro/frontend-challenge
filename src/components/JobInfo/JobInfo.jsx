import React from 'react';
import styled from 'styled-components';

function JobInfo({ postedAt, contract, location }) {
  return (
    <StyledJobInfo>
      <Info>{postedAt}</Info>
      <Info>{contract}</Info>
      <Info>{location}</Info>
    </StyledJobInfo>
  );
}

const Info = styled.li`
  color: rgba(133, 139, 139, 0.75);
  font-weight: 400;
  white-space: nowrap;

  &:first-child::before {
    font-size: 1.7em;
    content: ' ';
  }

  &:not(:first-child)::before {
    content: '.';
    display: inline-block;
    transform: translateY(-16%);
    padding-left: 4px;
    padding-right: 4px;
    font-size: 1.7em;

    @media (min-width: 550px) {
      padding-left: 8px;
      padding-right: 8px;
    }
  }
`;

const StyledJobInfo = styled.ul`
  display: flex;
  list-style: none;
  margin-bottom: 15px;
  @media (min-width: 491px) {
    margin-bottom: 0px;
  }
`;

export default JobInfo;
