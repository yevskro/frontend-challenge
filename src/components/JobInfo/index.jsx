import React from 'react';
import styled from 'styled-components';

function JobInfo({ postedAt, contract, location }) {
  return (
    <StyledJobInfo>
      <Info>{postedAt}</Info>
      <Info>{contract}</Info>
      <Info>{location}</Info>
    </StyledJobInfo>
  );
}

const Info = styled.li`
  white-space: nowrap;

  /* increasing the size of the bullet represented by content character '.' */
  &:first-child::before {
    font-size: 1.7rem;
    content: ' ';
  }

  &:not(:first-child)::before {
    content: '.';
    display: inline-block;
    transform: translateY(-16%);
    padding-left: 4px;
    padding-right: 4px;
    font-size: 1.7rem;

    @media (min-width: 600px) {
      padding-left: 8px;
      padding-right: 8px;
    }
  }
  @media (min-width: 600px) {
    font-size: ${({ theme }) => theme.fontSize.medium};
  }
`;

const StyledJobInfo = styled.ul`
  display: flex;
  list-style: none;
  color: ${({ theme }) => theme.color.neutral.darkGrayishCyan};
  font-weight: ${({ theme }) => theme.fontWeight.thin};
  font-size: ${({ theme }) => theme.fontSize.small};
`;

export default JobInfo;
