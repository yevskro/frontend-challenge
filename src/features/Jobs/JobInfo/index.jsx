import React from 'react';
import styled from 'styled-components';

function JobInfo({ postedAt, contract, location }) {
  return (
    <StyledJobInfo>
      <Info>{postedAt}</Info>
      <Info>{contract}</Info>
      <Info>{location}</Info>
    </StyledJobInfo>
  );
}

const Info = styled.li`
  white-space: nowrap; /* no wrap arounds to the next line */

  /* increasing the size of the bullet represented by content character '.' */
  &:first-child::before {
    /* 
      no bullet point before the first element fixing the font-size so the element
      does not jump
    */
    font-size: 1.7rem;
    content: ' ';
  }

  &:not(:first-child)::before {
    display: inline-block;
    content: '.';
    padding: 0px 4px 0px 4px;
    font-size: 1.7rem;
    transform: translateY(-16%); /* moves the . to the vertical center */

    @media (min-width: 600px) {
      /* increases the space between the point and the text around */
      padding: 0px 8px 0px 8px;
    }
  }

  @media (min-width: 600px) {
    /* works for tablet and desktop views */
    font-size: ${({ theme }) => theme.fontSize.medium};
  }
`;

const StyledJobInfo = styled.ul`
  display: flex;
  color: ${({ theme }) => theme.color.neutral.darkGrayishCyan};
  font-weight: ${({ theme }) => theme.fontWeight.thin};
  font-size: ${({ theme }) => theme.fontSize.small};
`;

export default JobInfo;
