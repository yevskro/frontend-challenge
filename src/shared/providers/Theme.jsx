import React from 'react';
import { ThemeProvider } from 'styled-components';
/* theme developed based on the style-guide */
const theme = {
  color: {
    primary: {
      desaturtedDarkCyan: 'hsl(180, 29%, 50%)',
    },
    neutral: {
      lightGrayishCyanBackground: 'hsl(180, 52%, 96%)',
      lightGrayishCyanFilterTablets: 'hsl(180, 31%, 95%)',
      darkGrayishCyan: 'hsl(180, 8%, 52%)',
      veryDarkGrayishCyan: 'hsl(180, 14%, 20%)',
    },
    shadow: 'rgba(94, 140, 160, 0.15)',
  },
  fontSize: {
    xxsmall: '0.7rem',
    xsmall: '0.8rem',
    small: '0.9rem',
    medium: '1rem',
    large: '1.2rem',
  },
  fontWeight: {
    thin: '400',
    normal: '500',
    bold: '600',
  },
  fontFamily: '"Spartan", "sans-serif"',
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
