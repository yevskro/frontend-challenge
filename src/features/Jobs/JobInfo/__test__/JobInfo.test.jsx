import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Theme from '../../../../shared/providers/Theme';
import JobInfo from '..';

test('renders', () => {
  render(
    <Theme>
      <JobInfo postedAt="2d ago" contract="Part Time" location="USA Only" />
    </Theme>
  );

  expect(screen.getByText('2d ago')).toBeInTheDocument();
  expect(screen.getByText('Part Time')).toBeInTheDocument();
  expect(screen.getByText('USA Only')).toBeInTheDocument();
});

test('snapshot', async () => {
  const tree = renderer
    .create(
      <Theme>
        <JobInfo newly featured>
          Durran
        </JobInfo>
      </Theme>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
