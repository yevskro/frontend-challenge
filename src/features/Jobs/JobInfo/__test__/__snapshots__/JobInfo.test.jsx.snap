// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c1 {
  white-space: nowrap;
}

.c1:first-child::before {
  font-size: 1.7rem;
  content: ' ';
}

.c1:not(:first-child)::before {
  display: inline-block;
  content: '.';
  padding: 0px 4px 0px 4px;
  font-size: 1.7rem;
  -webkit-transform: translateY(-16%);
  -ms-transform: translateY(-16%);
  transform: translateY(-16%);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: hsl(180,8%,52%);
  font-weight: 400;
  font-size: 0.9rem;
}

@media (min-width:600px) {
  .c1:not(:first-child)::before {
    padding: 0px 8px 0px 8px;
  }
}

@media (min-width:600px) {
  .c1 {
    font-size: 1rem;
  }
}

<ul
  className="c0"
>
  <li
    className="c1"
  />
  <li
    className="c1"
  />
  <li
    className="c1"
  />
</ul>
`;
